cmd/krel/cmd/patch-announce.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/release/pkg/log"
+	"k8s.io/release/pkg/patch"
+	"k8s.io/release/pkg/util"
+)
+
+// slap the subcommand onto the parent/root
+func init() {
+	cmd := patchAnnounceCommand()
+	rootCmd.AddCommand(cmd)
+}
+
+func patchAnnounceCommand() *cobra.Command {
+	opts := patch.AnnounceOptions{}
+
+	cmd := &cobra.Command{
+		Use:           "patch-announce",
+		Short:         "Send out patch release announcement mails",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Args:          cobra.MaximumNArgs(0), // no additional/positional args allowed
+	}
+
+	cobra.OnInitialize(initConfig) // ?
```

Comment:
this is a global setting for all cobra commands, which might confuse people debugging the code in the future.
why not instead execute the function on demand where needed?

Reply:
TL;DR: will remove that.

That's a thing I actually wanted to remove, seems I forgot about it. IIRC, it's an empty function right now anyway and used/called by all (sub)commands. It is a remnant of me trying to figure out how things are currently done.